Eastern-Research-Group/mywaterway
# feature/776 switch to vite

## Related Issues:
* [HMW-776](https://jira.epa.gov/browse/HMW-776)

## Main Changes:
* Switched to vite.
* Removed flow.
* Converted .js/.jsx to .ts/.tsx. 
  * There is still alot of work to do to convert to typescript.
  * I removed `tsc` from the build script, so we don't get a bunch of typescript errors during the build. We can continue switching to typescript in a separate PR or slowly as we touch files. Once we are happy with the conversion, we can add `tsc` back into the build script.
* Fixed some of the cypress tests.

## Steps To Test:
1. Run `npm run setup`
2. Run `npm run dev`
3. Try to test as many features as possible
4. Run `npm run full_coverage`.
    * You can temporarily delete most of the cypress spec files so this will complete more quickly. This is mainly here to verify that coverage still works.

## TODO:
- [ ] Continue switching to TypeScript.
- [ ] When we are happy with our work on TypeScript, update the build script in client/package.json so that it is `tsc && vite build`, instead of `vite build`.
